### ISSUE TITLE
Indent 4 with VariableDeclarator and ObjectExpression set as "first" it fails for Object literal declaration

### ISSUE THREAD

**Issue by saptaru** [eslint/eslint#13524]
**Tell us about your environment**

* **ESLint Version: 7.3.1**
* **Node Version: v10.15.3**
* **npm Version: 6.14.5**

**What parser (default, Babel-ESLint, etc.) are you using?**
webpack + babel-loader

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    'env'          : {
        'browser'     : true,
        'es2020'      : true,
        'amd'         : true,
        'node'        : true,
        'jest/globals': true
    },
    'extends'      : [
        'eslint:recommended',
        'plugin:vue/essential'
    ],
    'parserOptions': {
        'ecmaVersion': 11,
        'sourceType' : 'module'
    },
    'plugins'      : [
        'vue',
        'jest'
    ],
    'globals'      : {
        '$GLOBAL_VARS'           : 'readonly',
        '__webpack_public_path__': 'writable',
        '$publicPath'            : 'readonly'
    },
    'rules'        : {
        'indent'                 : [
            'warn',
            4,
            {
                'VariableDeclarator': 'first',
                'ObjectExpression'  : 'off'
            }
        ],
        'brace-style'            : 'warn',
        'eol-last'               : 'warn',
        'no-multiple-empty-lines': [
            'warn',
            {
                'max'   : 1,
                'maxEOF': 1,
                'maxBOF': 0
            }
        ],
        'newline-before-return'  : 'warn',
        'newline-after-var'      : [
            'warn',
            'always'
        ],
        'lines-around-directive' : [
            'warn',
            'always'
        ],
        'space-before-blocks'    : 'warn',
        'no-unused-vars'         : 'warn',

        // 'vue/script-indent'       : [
        //     'warn',
        //     4,
        //     {
        //         'baseIndent': 1
        //     }
        // ],
        'newline-per-chained-call': [
            1
        ]
    },
    'overrides'    : [
        {
            'files': ['*.vue'],
            'rules': {
                'indent': 'off'
            }
        }
    ]
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/* eslint indent: ["error", 4, { "VariableDeclarator": "first", "ObjectExpression": "first" }] */
describe('test1', () => {
    describe('test1.1', () => {
        test('with fail reason', () => {
            const myvariable = {
                      [STEPS]: {
                          code : 123,
                          index: 4
                      }
                  },
                  cart       = {
                      failReason: 'failReason'
                  };
        });
    });
});

```

```bash
npx eslint --ext .js,.vue ./some-folder
```

**What did you expect to happen?**
There should be an option to align the literal object or content from a function starting with the first variable declaration. The example above I expect to make it work. The reason for this is that with my actual jetBrains configuration I cannot set it to be 100% like in the eslint configuration.

**What actually happened? Please include the actual, raw output from ESLint.**
You can have an example of the issue here: https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGluZGVudDogW1wiZXJyb3JcIiwgNCwgeyBcIlZhcmlhYmxlRGVjbGFyYXRvclwiOiBcImZpcnN0XCIsIFwiT2JqZWN0RXhwcmVzc2lvblwiOiBcImZpcnN0XCIgfV0gKi9cbmRlc2NyaWJlKCd0ZXN0MScsICgpID0+IHtcbiAgICBkZXNjcmliZSgndGVzdDEuMScsICgpID0+IHtcbiAgICAgICAgdGVzdCgnd2l0aCBmYWlsIHJlYXNvbicsICgpID0+IHtcbiAgICAgICAgICAgIGNvbnN0IG15dmFyaWFibGUgPSB7XG4gICAgICAgICAgICAgICAgICAgICAgW1NURVBTXToge1xuICAgICAgICAgICAgICAgICAgICAgICAgICBjb2RlIDogMTIzLFxuICAgICAgICAgICAgICAgICAgICAgICAgICBpbmRleDogNFxuICAgICAgICAgICAgICAgICAgICAgIH1cbiAgICAgICAgICAgICAgICAgIH0sXG4gICAgICAgICAgICAgICAgICBjYXJ0ICAgICAgID0ge1xuICAgICAgICAgICAgICAgICAgICAgIGZhaWxSZWFzb246ICdmYWlsUmVhc29uJ1xuICAgICAgICAgICAgICAgICAgfTtcbiAgICAgICAgfSk7XG4gICAgfSk7XG59KTtcbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6OSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==

Or if I disable the ObjectExpression then it will be like this: https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGluZGVudDogW1wiZXJyb3JcIiwgNCwgeyBcIlZhcmlhYmxlRGVjbGFyYXRvclwiOiBcImZpcnN0XCIsIFwiT2JqZWN0RXhwcmVzc2lvblwiOiBcIm9mZlwiIH1dICovXG5kZXNjcmliZSgndGVzdDEnLCAoKSA9PiB7XG4gICAgZGVzY3JpYmUoJ3Rlc3QxLjEnLCAoKSA9PiB7XG4gICAgICAgIHRlc3QoJ3dpdGggZmFpbCByZWFzb24nLCAoKSA9PiB7XG4gICAgICAgICAgICBjb25zdCBteXZhcmlhYmxlID0ge1xuICAgICAgICAgICAgICAgICAgICAgIFtTVEVQU106IHtcbiAgICAgICAgICAgICAgICAgICAgICAgICAgY29kZSA6IDEyMyxcbiAgICAgICAgICAgICAgICAgICAgICAgICAgaW5kZXg6IDRcbiAgICAgICAgICAgICAgICAgICAgICB9XG4gICAgICAgICAgICAgICAgICB9LFxuICAgICAgICAgICAgICAgICAgY2FydCAgICAgICA9IHtcbiAgICAgICAgICAgICAgICAgICAgICBmYWlsUmVhc29uOiAnZmFpbFJlYXNvbidcbiAgICAgICAgICAgICAgICAgIH07XG4gICAgICAgIH0pO1xuICAgIH0pO1xufSk7XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjksInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=
What it is funny is that in the second example line no. 13 it is not giving any errors but line no. 10 it is.

I think this is related to: https://github.com/eslint/eslint/issues/11265

**Are you willing to submit a pull request to fix this bug?**
No

**Comment by anikethsaha** [comment]
Thanks for the report. 

I think it is working fine. `VariableDeclarator: first` is working as expected. the second (and rest ) element should indented as same as the first one. 

`"ObjectExpression": "first"` is working as expected as well. the property are aligned fine with the first prop. 

if you want to have the second variable indented same as  `}`, then I think you can remove `VariableDeclarator`

```js
/* eslint indent: ["error", 4, { "ObjectExpression": "first" }] */
describe('test1', () => {
    describe('test1.1', () => {
        test('with fail reason', () => {
            const myvariable = {
                    [STEPS]: {
                        code : 123,
                        index: 4
                    }
                },
                cart       = {
                    failReason: 'failReason'
                };
        });
    });
});
```
 

let me know if I am missing anything.

**Comment by saptaru** [comment]
Thanks for the response,

If you are looking at my first example https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGluZGVudDogW1wiZXJyb3JcIiwgNCwgeyBcIlZhcmlhYmxlRGVjbGFyYXRvclwiOiBcImZpcnN0XCIsIFwiT2JqZWN0RXhwcmVzc2lvblwiOiBcImZpcnN0XCIgfV0gKi9cbmRlc2NyaWJlKCd0ZXN0MScsICgpID0+IHtcbiAgICBkZXNjcmliZSgndGVzdDEuMScsICgpID0+IHtcbiAgICAgICAgdGVzdCgnd2l0aCBmYWlsIHJlYXNvbicsICgpID0+IHtcbiAgICAgICAgICAgIGNvbnN0IG15dmFyaWFibGUgPSB7XG4gICAgICAgICAgICAgICAgICAgICAgW1NURVBTXToge1xuICAgICAgICAgICAgICAgICAgICAgICAgICBjb2RlIDogMTIzLFxuICAgICAgICAgICAgICAgICAgICAgICAgICBpbmRleDogNFxuICAgICAgICAgICAgICAgICAgICAgIH1cbiAgICAgICAgICAgICAgICAgIH0sXG4gICAgICAgICAgICAgICAgICBjYXJ0ICAgICAgID0ge1xuICAgICAgICAgICAgICAgICAgICAgIGZhaWxSZWFzb246ICdmYWlsUmVhc29uJ1xuICAgICAgICAgICAgICAgICAgfTtcbiAgICAgICAgfSk7XG4gICAgfSk7XG59KTtcbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6OSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==
and then go to the "Fixed code" tab you will see that the line no 10 is not aligned with line no 13.

From what I can tell, eslint in current configuration, for object literal properties adds indentations from the start of the line, instead jetbrains adds indentations from first variable declaration. My question is in fact how can I achieve the same formatting provided in the example with eslint?

**Comment by anikethsaha** [comment]
the closing braces are matching the indentation of the opening brace line. 

so as you are using `"VariableDeclarator": "first"` , the second variable is in align with the first one. But first one does have a `const` keyword, so the indentation of calc is beginning from `const` not from the variable name. 
So the closing `}` will match with the indentation of the line of the opening brace. so that is matching with the beginning of the `const`. 
But in the second variable, the opening the brace line `{` indentation is starting from `cart` so the closing brace `}` will be indented with the `cart` not with the `const`. 

for example if you do like this 

```js
   cart = 
           {
                      failReason: 'failReason'
     };
```

It will match with the indentation of the `{` not with the `cart`

![image](https://user-images.githubusercontent.com/26347874/88408047-a27a7180-cdf0-11ea-8b29-beca56a25c0e.png)

Now if you remove the `const` , it will match with the indentation of the `myvariable`

[see the `fixed code`](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGluZGVudDogW1wiZXJyb3JcIiwgNCwgeyBcIlZhcmlhYmxlRGVjbGFyYXRvclwiOiBcImZpcnN0XCIsIFwiT2JqZWN0RXhwcmVzc2lvblwiOiBcImZpcnN0XCIgfV0gKi9cbmRlc2NyaWJlKCd0ZXN0MScsICgpID0+IHtcbiAgICBkZXNjcmliZSgndGVzdDEuMScsICgpID0+IHtcbiAgICAgICAgdGVzdCgnd2l0aCBmYWlsIHJlYXNvbicsICgpID0+IHtcbiAgICAgICAgICAgICAgbXl2YXJpYWJsZSA9IHtcbiAgICAgICAgICAgICAgICAgICAgICBbU1RFUFNdOiB7XG4gICAgICAgICAgICAgICAgICAgICAgICAgIGNvZGUgOiAxMjMsXG4gICAgICAgICAgICAgICAgICAgICAgICAgIGluZGV4OiA0XG4gICAgICAgICAgICAgICAgICAgICAgfVxuICAgICAgICAgICAgICAgICAgfSxcbiAgICAgICAgICAgICAgICAgIGNhcnQgPSB7XG4gICAgICAgICAgICAgICAgICAgICAgZmFpbFJlYXNvbjogJ2ZhaWxSZWFzb24nXG4gICAgICAgICAgICAgICAgICB9O1xuICAgICAgICB9KTtcbiAgICB9KTtcbn0pO1xuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo5LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

If you remove the `VariableDeclarator: first` if will now match with indent `4` (for second variable) [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGluZGVudDogW1wiZXJyb3JcIiwgNCwgeyAgXCJPYmplY3RFeHByZXNzaW9uXCI6IFwiZmlyc3RcIiB9XSAqL1xuZGVzY3JpYmUoJ3Rlc3QxJywgKCkgPT4ge1xuICAgIGRlc2NyaWJlKCd0ZXN0MS4xJywgKCkgPT4ge1xuICAgICAgICB0ZXN0KCd3aXRoIGZhaWwgcmVhc29uJywgKCkgPT4ge1xuICAgICAgICAgICAgICBjb25zdCBteXZhcmlhYmxlID0ge1xuICAgICAgICAgICAgICAgICAgICAgIFtTVEVQU106IHtcbiAgICAgICAgICAgICAgICAgICAgICAgICAgY29kZSA6IDEyMyxcbiAgICAgICAgICAgICAgICAgICAgICAgICAgaW5kZXg6IDRcbiAgICAgICAgICAgICAgICAgICAgICB9XG4gICAgICAgICAgICAgICAgICB9LFxuICAgICAgICAgICAgICAgICAgICAgY2FydCA9IHtcbiAgICAgICAgICAgICAgICAgICAgICBmYWlsUmVhc29uOiAnZmFpbFJlYXNvbidcbiAgICAgICAgICAgICAgICAgIH07XG4gICAgICAgIH0pO1xuICAgIH0pO1xufSk7XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjksInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

Though I agree, it does look a bit weird like this 

![image](https://user-images.githubusercontent.com/26347874/88408465-38160100-cdf1-11ea-857a-97c83f5d7a7d.png)

I think this change of aligning the `}` with the beginning of `cart` (second variable) will be an `enhancement` maybe instead of `bug` ! and according to the new [policy](https://eslint.org/blog/2020/05/changes-to-rules-policies) enhancement for stylistic rules are frozen. 

Lets hear from others what they thinks
